### PR TITLE
enabling test for comgr

### PR DESCRIPTION
**Pull request by afzpatel**
Command for install+test:
**spack install --verbose --test=root comgr**  

Log:
[install-time-test-log.txt](https://github.com/spack/spack/files/10797845/install-time-test-log.txt)

Edit:
Updated Log using ctest for testing:
[install-time-test-log.txt](https://github.com/spack/spack/files/10817057/install-time-test-log.txt)

### CODE REVIEW

File: var/spack/repos/builtin/packages/comgr/package.py
Comment (haampie) [inline var/spack/repos/builtin/packages/comgr/package.py]: Can you write this in Python? Edit: actually I have no clue what `run_test` is. Can't you just run `ctest`? It's a cmake package after all.

Comment (afzpatel) [reply var/spack/repos/builtin/packages/comgr/package.py]: Regarding `run_test,` I used it here because all the rocm components I've seen that currently have spack testing enabled(hipblas, rocblas, rocfft, etc.), all use `run_test` for running the test binaries.

Comment (afzpatel) [reply var/spack/repos/builtin/packages/comgr/package.py]: I removed the whole `check` function and it now uses ctest for testing.

**Review by srekolam — APPROVED** [review]

**Review by renjithravindrankannath — APPROVED** [review]
LGTM..

**Comment by afzpatel** [comment]
@haampie I have updated the code since your last comment. Please kindly review the changes when you get the chance.

**Comment by afzpatel** [comment]
@haampie

**Comment by afzpatel** [comment]
@haampie

**Comment by afzpatel** [comment]
Hi @tldahlgren this pr is ready to be reviewed by your team

**Comment by afzpatel** [comment]
Hello @haampie @tldahlgren @alalazo this pr is waiting for final approval

**Review by alalazo — APPROVED** [review]

**Pull request merged**
